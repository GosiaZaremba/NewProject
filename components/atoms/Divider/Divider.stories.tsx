import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { Divider } from './Divider';
import { Colors } from '../../../constants/colors';

storiesOf('atoms/Divider', module)
    .addDecorator((story) => story())
    .add('default', () => <Divider text={'OR'} />);
