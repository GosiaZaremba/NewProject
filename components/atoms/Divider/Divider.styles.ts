import {StyleSheet} from 'react-native';
import {Colors} from '../../../constants/colors';

export default StyleSheet.create({
  outerContainer: {
    borderWidth: 1,
    borderColor: Colors.grey,
    width: '100%',
    height: 1,
    borderRadius: 5,
    position: 'relative',
    alignItems: 'center',
    justifyContent: 'center',
  },
  innerContainer: {
    borderWidth: 1,
    borderColor: Colors.grey,
    padding: 5,
    borderRadius: 5,
    position: 'absolute',
    backgroundColor: Colors.light,
  },
});
