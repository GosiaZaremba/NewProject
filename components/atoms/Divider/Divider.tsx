import React from 'react';
import { Text, View } from 'react-native';
import styles from './Divider.styles';

export type Props = {
    text: string;
    testID?: string;
};

export const Divider: React.FC<Props> = ({ text, testID }) => {
    return (
        <View style={styles.outerContainer}>
            <View testID={testID} style={styles.innerContainer}>
                <Text>{text}</Text>
            </View>
        </View>
    );
};
