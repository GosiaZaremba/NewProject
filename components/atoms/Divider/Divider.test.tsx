import { render } from '@testing-library/react-native';
import React from 'react';
import '@testing-library/jest-native';
import renderer from 'react-test-renderer';

import 'react-native';
import { Divider } from './Divider';

describe('Atom Divider', () => {
    test('Component Snapshot', () => {
        const tree = renderer
            .create(<Divider text={'Divider text'} testID={'atom-divider'} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('ensure variants works correctly', () => {
        const { getByTestId } = render(
            <Divider text={'Divider text'} testID={'atom-divider'} />
        );

        const divider = getByTestId('atom-divider');
        expect(divider).toHaveTextContent('Divider text');
    });
});
