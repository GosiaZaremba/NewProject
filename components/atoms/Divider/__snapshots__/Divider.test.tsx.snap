// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom Divider Component Snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#908D8E",
      "borderRadius": 5,
      "borderWidth": 1,
      "height": 1,
      "justifyContent": "center",
      "position": "relative",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#F5F0F2",
        "borderColor": "#908D8E",
        "borderRadius": 5,
        "borderWidth": 1,
        "padding": 5,
        "position": "absolute",
      }
    }
    testID="atom-divider"
  >
    <Text>
      Divider text
    </Text>
  </View>
</View>
`;
